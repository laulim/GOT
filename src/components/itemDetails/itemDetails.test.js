import React from 'react';
import ItemDetails from './itemDetails';
import {shallow} from 'enzyme';

describe('Testing <ItemDetails/>', () => {
    const itemD = shallow(<ItemDetails/>);

    describe('Testing snap & state', () => {
        it('ItemDetails have rendered correctly', () => {
            expect(itemD).toMatchSnapshot();
        });

        it('ItemDetails state "item" is null', () => {
            expect(itemD.state().item).toBeNil();
        });

        it('ItemDetails state "loading" is false', () => {
            expect(itemD.state().loading).toBeFalsy();
        });

        it('ItemDetails state "error" is false', () => {
            expect(itemD.state().error).toBeFalsy();
        });
    });

    describe('Hendlers tests', () => {
      it('Testing updateItem', () => {
        itemD.instance().updateItem();
        expect(itemD.state().loading).toBeFalsy();
      });

      it('Testing onError', () => {
        itemD.instance().onError();
        expect(itemD.state().loading).toBeFalsy();
        expect(itemD.state().error).toBeTruthy();
      });
    });
});