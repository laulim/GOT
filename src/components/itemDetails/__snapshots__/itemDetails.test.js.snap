// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ItemDetails/> Testing snap & state ItemDetails have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <span
          className="select-error"
        >
          Please select a undefined
        </span>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a undefined",
          "className": "select-error",
        },
        "ref": null,
        "rendered": "Please select a undefined",
        "type": "span",
      },
      null,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  background-color: #fff;
  padding: 25px 25px 15px 25px;
  margin-bottom: 40px;
  border-radius: 0.25rem;
  h4 {
    margin-bottom: 20px;
    text-align: center;
  }
  .select-error {
    color: #000;
    text-align: center;
    font-size: 26px;
  }
  .term {
    font-weight: bold;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <span
            className="select-error"
          >
            Please select a undefined
          </span>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please select a undefined",
            "className": "select-error",
          },
          "ref": null,
          "rendered": "Please select a undefined",
          "type": "span",
        },
        null,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  background-color: #fff;
  padding: 25px 25px 15px 25px;
  margin-bottom: 40px;
  border-radius: 0.25rem;
  h4 {
    margin-bottom: 20px;
    text-align: center;
  }
  .select-error {
    color: #000;
    text-align: center;
    font-size: 26px;
  }
  .term {
    font-weight: bold;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
